src/transformers/models/patchtst/modeling_patchtst.py

```diff
@@ -0,0 +1,1958 @@
+            across channels. Defaults to True.
+        mask_value (int, optional): Value to use for masking. Defaults to 0.
+        seed_number (int, optional): Random seed, when None seed is not set. Defaults to None.
+    """
+
+    def __init__(
+        self,
+        mask_type: str = "random",
+        mask_ratio=0.5,
+        mask_patches: list = [2, 3],
+        mask_patch_ratios: list = [1, 1],
+        channel_consistent_masking: bool = False,
+        unmasked_channel_indices: list = None,
+        mask_value=0,
+        seed_number: Optional[int] = None,
+    ):
+        # if seed_number:
+        #     set_seed(seed_number)
+        self.mask_ratio = mask_ratio
+        self.channel_consistent_masking = channel_consistent_masking
+        self.mask_type = mask_type
+        self.mask_patches = mask_patches
+        self.mask_patch_ratios = mask_patch_ratios
+        self.unmasked_channel_indices = unmasked_channel_indices
+        self.mask_value = mask_value
+        if self.unmasked_channel_indices is not None:
+            self.unmasked_channel_indices.sort()
+        self.seed_number = seed_number
+
+        super().__init__()
+
+    def forward(self, x: torch.Tensor):
+        """
+        Input:
+            x: patched input
+                4D: [bs x num_input_channels x num_patches x patch_length]
+
+        Output:
+            x_mask: Masked patched input
+                4D: [bs x num_input_channels x num_patches x patch_length]
+            mask: bool tensor indicating True on masked points
+                4D: [bs x num_input_channels x num_patch]
+        """
+
+        if self.mask_type == "random":
+            x_mask, mask = random_masking(
+                xb=x,
+                mask_ratio=self.mask_ratio,
+                unmasked_channel_indices=self.unmasked_channel_indices,
+                channel_consistent_masking=self.channel_consistent_masking,
+                mask_value=self.mask_value,
+                seed_number=self.seed_number,
+            )
+        elif self.mask_type == "forecast":
+            x_mask, mask = forecast_masking(
+                xb=x,
+                patch_lengths=self.mask_patches,
+                mix_ratio=self.mask_patch_ratios,
+                unmasked_channel_indices=self.unmasked_channel_indices,
+                mask_value=self.mask_value,
+                seed_number=self.seed_number,
+            )
+        else:
+            raise Exception("Invalid mask type")
+
+        mask = mask.bool()  # mask: [bs x num_input_channels x num_patch]
+
+        return x_mask, mask
+
+
+class ChannelAttentionTSTEncoder(nn.Module):
+    def __init__(self, config: PatchTSTConfig):
+        super().__init__()
+
+        self.layers = nn.ModuleList([ChannelAttentionTSTEncoderLayer(config) for i in range(config.encoder_layers)])
+
+    def forward(self, src: torch.Tensor, output_hidden_states: Optional[bool] = None):
```

Comment:
for consistency let's use hidden_state instead of src

Reply:
Yes, it is fixed